ext/zip/php_zip.c

```diff
@@ -3006,6 +3016,9 @@ PHP_METHOD(ZipArchive, isCompressionMethodSupported)
 	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l|b", &method, &enc) == FAILURE) {
 		return;
 	}
+	if (method < 0) {
+		RETURN_FALSE;
+	}
 	RETVAL_BOOL(zip_compression_method_supported((zip_int32_t)method, enc));
```

Comment:
Isn't this a problem if `method` is larger than a 32bit int but still positive on x64?